src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
Let's try to use `.reduce` instead of `.forEach`.
`.find` can possibly return undefined we need to handle this scenario.
```suggestion
const sum = children.reduce((acc, child) => {
  const mother = people.find(m => m.name === child.mother);
  return mother ? acc + child.born - mother.born : acc;
}, 0);
```